packages/fiber/src/core/renderer.ts

```diff
@@ -243,7 +243,7 @@ function createRenderer<TCanvas>(roots: Map<TCanvas, Root>) {
       invalidateInstance(instance)
     })
 
-    if (rootState.internal && instance.raycast && prevHandlers !== localState.eventCount) {
+    if (localState.parent && rootState.internal && instance.raycast && prevHandlers !== localState.eventCount) {
       // Pre-emptively remove the instance from the interaction manager
       const index = rootState.internal.interaction.indexOf(instance as unknown as THREE.Object3D)
       if (index > -1) rootState.internal.interaction.splice(index, 1)
```

Comment:
Are instances' handlers cleaned up when they're unmounted?

Reply:
a handler is just a function stored in an array (state.internal.interaction) if it's removed from there it's out.